### PR TITLE
fix: we calculate event and property counts infrequently as a thundering herd

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

We only process each team once every day at most. So, investigating errors is slow and frustrating.

## Changes

Uses the approach in #13209 to let us schedule 300 teams every five minutes for calculation
Instead of running thousands of teams as fast as possible every 6 hours
We now run 1 team at a time

Stops the counter from depending on the model classes so we can write some tests on its behaviour

## How did you test this code?

Ran it locally and saw teams updated as expected

### CODE REVIEW

**Comment by pauldambra** [comment]
has an error in Cypress which looks real but I think also unrelated to this :/

```
2022-12-11T11:26:15.234154Z [info     ] request_started                [django_structlog.middlewares.request] ip=127.0.0.1 pid=21 request=<WSGIRequest: PATCH '/api/projects/1/insights/10/'> request_id=3db41e5a-d941-4147-bfd7-1f67bf79eac5 tid=140630661770032 user_agent=Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/10.11.0 Chrome/106.0.5249.51 Electron/21.0.0 Safari/537.36
2022-12-11T11:26:15.287086Z [info     ] task_enqueued                  [django_structlog.celery.receivers] child_task_id=0f7f5da2-2b34-4502-99de-3359ab83b58f child_task_name=posthog.celery.sync_insight_caching_state ip=127.0.0.1 pid=21 request_id=3db41e5a-d941-4147-bfd7-1f67bf79eac5 tid=140630661770032
2022-12-11T11:26:15.313037Z [error    ] update or delete on table "posthog_dashboardtile" violates foreign key constraint "posthog_insightcachi_dashboard_tile_id_95935ecf_fk_posthog_d" on table "posthog_insightcachingstate"
DETAIL:  Key (id)=(10) is still referenced from table "posthog_insightcachingstate".
 [posthog.exceptions] ip=127.0.0.1 path=/api/projects/1/insights/10/ pid=21 request_id=3db41e5a-d941-4147-bfd7-1f67bf79eac5 tid=140630661770032
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/base.py", line 242, in _commit
    return self.connection.commit()
psycopg2.errors.ForeignKeyViolation: update or delete on table "posthog_dashboardtile" violates foreign key constraint "posthog_insightcachi_dashboard_tile_id_95935ecf_fk_posthog_d" on table "posthog_insightcachingstate"
DETAIL:  Key (id)=(10) is still referenced from table "posthog_insightcachingstate".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/mixins.py", line 82, in partial_update
    return self.update(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/mixins.py", line 68, in update
    self.perform_update(serializer)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/mixins.py", line 78, in perform_update
    serializer.save()
  File "/usr/local/lib/python3.8/site-packages/rest_framework/serializers.py", line 200, in save
    self.instance = self.update(self.instance, validated_data)
  File "/code/posthog/api/insight.py", line 284, in update
    self._update_insight_dashboards(dashboards, instance)
  File "/code/posthog/api/insight.py", line 326, in _update_insight_dashboards
    instance.dashboards.set(changes_to_apply, clear=True)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/fields/related_descriptors.py", line 1015, in set
    self.add(*new_objs, through_defaults=through_defaults)
  File "/usr/local/lib/python3.8/site-packages/django/db/transaction.py", line 246, in __exit__
    connection.commit()
  File "/usr/local/lib/python3.8/site-packages/django/utils/asyncio.py", line 33, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/base.py", line 266, in commit
    self._commit()
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/base.py", line 242, in _commit
    return self.connection.commit()
  File "/usr/local/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/base.py", line 242, in _commit
    return self.connection.commit()
django.db.utils.IntegrityError: update or delete on table "posthog_dashboardtile" violates foreign key constraint "posthog_insightcachi_dashboard_tile_id_95935ecf_fk_posthog_d" on table "posthog_insightcachingstate"
DETAIL:  Key (id)=(10) is still referenced from table "posthog_insightcachingstate".
```

**Comment by pauldambra** [comment]
see #13256

**Comment by macobo** [comment]
Oof, I'm scared about merging this since I'm not sure whether this will slow down the CH cluster further. Let's sync and sell me on the idea!

**Comment by pauldambra** [comment]
Your response in the customer thread https://posthog.slack.com/archives/C02EZ1PMYPK/p1670842245871569?thread_ts=1669753623.587739&cid=C02EZ1PMYPK

Also made me realise that this view is confusing...

We say "query 30 day usage" but we present "number of insights created in the last 30 days that use X" when a much more reasonable interpretation is "number of times a query _ran_ in the last 30 days that use X"

Screenshot 2022-12-12 at 11 06 13

File: posthog/celery.py [hunk 1]
```diff
@@ -510,6 +535,14 @@ def update_cache_task(caching_state_id: UUID):
     update_cache(caching_state_id)
 
 
+@app.task(ignore_result=True, bind=True)
+@single_executing_at_once("celery_update_cache_task")
```
Comment (pauldambra) [inline posthog/celery.py:539]:
```suggestion
@single_executing_at_once("calculate_event_property_usage_for_team_task")
```

**Comment by pauldambra** [comment]
@macobo I changed the number being enqueued to be an instance setting so we can experiment with this.

Currently we enqueue every team that hasn't been processed in the last 24 hours once every six hours. So, we're processing thousands of teams in a short period of time.

**Comment by pauldambra** [comment]
let's close this and come back to this area properly with some focus

**Pull request closed without merging**